src/Command/UnregisterAdherentsCommand.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace AppBundle\Command;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Unregistration;
+use AppBundle\Membership\AdherentRegistry;
+use AppBundle\Repository\AdherentRepository;
+use Doctrine\ORM\EntityManager;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Exception\FileNotFoundException;
+
+class UnregisterAdherentsCommand extends ContainerAwareCommand
+{
+    const COMMAND_NAME = 'app:adherent:unregister';
+
+    /**
+     * @var EntityManager
+     */
+    private $em;
+
+    /**
+     * @var AdherentRepository
+     */
+    private $adherentRepository;
+
+    /**
+     * @var AdherentRegistry
+     */
+    private $adherentRegistry;
+
+    private $notFoundEmails = [];
+
+    private $hosts = [];
+
+    private $citizenProjectAdministrators = [];
+
+    protected function configure()
+    {
+        $this
+            ->setName(self::COMMAND_NAME)
+            ->addArgument('fileUrl', InputArgument::REQUIRED)
+            ->setDescription('Unregister adherents from CSV file (only the fifth column "email" is taken into account)');
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->em = $this->getContainer()->get('doctrine.orm.entity_manager');
+        $this->adherentRepository = $this->em->getRepository(Adherent::class);
+        $this->adherentRegistry = $this->getContainer()->get(AdherentRegistry::class);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        try {
+            $emails = $this->parseCSV($input->getArgument('fileUrl'), 'r');
+        } catch (FileNotFoundException $exception) {
+            $output->writeln($exception->getMessage());
+
+            return 1;
+        }
+
+        $this->em->beginTransaction();
+
+        $this->unregisterAdherents($emails, $output);
+
+        $this->em->commit();
+
+        $output->writeln(PHP_EOL.'Unregistration finished.');
+
+        if ($this->notFoundEmails) {
+            $output->writeln('The following email addresses were not found in DB :');
+            foreach ($this->notFoundEmails as $email) {
+                $output->writeln($email);
+            }
+        }
```

Comment:
Just a little note, can you add an empty line after a `if` condition ?

Reply:
Done